src/gl/directx11/vcTexture.cpp

```diff
@@ -470,11 +470,21 @@ bool vcTexture_EndReadPixels(vcTexture *pTexture, uint32_t x, uint32_t y, uint32
 
   UD_ERROR_IF(g_pd3dDeviceContext->Map(pStagingTexture, 0, D3D11_MAP_READ, 0, &msr) != S_OK, udR_InternalError);
 
-  pPixelData = ((uint32_t *)msr.pData) + (x + y * pTexture->width);
-  for (int i = 0; i < (int)height; ++i)
+  if (x == 0 && y == 0 && width == pTexture->width && height == pTexture->height)
+  {
+    pPixelData = (uint32_t*)msr.pData;
+    memcpy((uint8_t*)pPixels, pPixelData, height * width * pixelBytes);
+  }
+  else
   {
-    memcpy((uint8_t*)pPixels + (i * pTexture->width), pPixelData, width * pixelBytes);
-    pPixelData += pTexture->width;
+    //TODO: This block is bugged. Does not do a complete copy.
+
+    pPixelData = ((uint32_t *)msr.pData) + (x + y * pTexture->width);
+    for (int i = 0; i < (int)height; ++i)
+    {
+      memcpy((uint8_t*)pPixels + (i * pTexture->width), pPixelData, width * pixelBytes);
```

Comment:
`pPixels` is a `void*` unlike `pPixelData` which is a `uint32_t*`, so this add here is probably missing a `4 *`?

Reply:
You are correct, that fixes it. I'll leave the optimised route anyway- I imagine that will be the much more common case.